### ISSUE TITLE
[Feature Request] broadcast an intent during a flow

### ISSUE THREAD

**Issue by andre-artus** [mobile-dev-inc/maestro#1142]
**Is your feature request related to a problem? Please describe.**
Some of my use cases are initiated from a broadcast  (i.e. implements a BroadcastReceiver). I would like to automate the tests for these receivers.  

**Describe the solution you'd like**
Some mechanism to initiate or simulate a broadcast (Local or System), or call adb to do it and check the result.
I would like to test when the following are sent:
 - correct broadcast (data/params)
 - incorrect broadcast ,
 - multiple concurrent broadcast (attempt to overload)
 Also
 - no ANR 
 - What happens when app is in non-interrupt state (e.g. user filling in a form) and intent is broadcast
- Behaviour under Doze, standby or process death.
- Test order of broadcast receipt (if dependent on order).

I'm sure I'm leaving something out, but that should cover a lot of it.

**Describe alternatives you've considered**
I'm currently just using scripts calling into ADB with a variety of known values and logging the interactions.

**Comment by amanjeetsingh150** [comment]
Hey @andre-artus can you explain the detailed use case you are looking at? What type of behavior do you want to verify in the maestro test?

**Comment by MaaxGr** [comment]
We use broadcast Intents to simulate the behaviour of the physical scan engine on enterprise devices (as Zebra TC58) on emulators:
![image](https://github.com/mobile-dev-inc/maestro/assets/43404580/c53f725c-753c-4bde-915e-c47e44b1ccbd)

Example: Send Barcode "test" to our application:
adb shell am broadcast -a de.guettlerlogistik.hallenapp.scan --es barcode "test" de.guettlerlogistik.hallenapp

That should also be supported with maestro. :)

Have a great week.

Max

**Comment by andre-artus** [comment]
Sorry, @amanjeetsingh150 , I failed to respond to you (I don;t know if I missed the notification) but Max's examples is very similar to ours (scanner or other accessories on enterprise devices).

**Comment by MaaxGr** [comment]
@andre-artus How did you solved it with scripts calling into ADB. Can you share an example?

**Comment by bartekpacia** [comment]
Hey all,

This seems to a bit of a more niche use case, and we're a small team busy with work, so I cannot share any timeline on when we will implement it. That said if anyone decided to contribute this feature, we're always happy to review PRs.

**Comment by LukStankovic** [comment]
Yes, this feature would be extremely useful for my workflow too, as I need to test an application on a Zebra scanner 🙏🏻
